metric.go

```diff
@@ -136,27 +113,26 @@ func (m *MetricDefinition) NameWithTags() string {
 		return m.nameWithTags
 	}
 
-	sort.Strings(m.Tags)
+	nameWithTagsBuffer := &bytes.Buffer{}
+	writeSortedTagString(nameWithTagsBuffer, m.Name, m.Tags)
```

Comment:
Are we going to check for the returned error here or not?

Reply:
so, this is a bit funny. the only way how `writeSortedTagString()` would return an error would be if the first argument is an `io.Writer` that returned an error either from `Write()` or from `WriteString()`. Now, according to their signature, they can return an error. But in our code we only use 3 types of `io.Writer`, which are:
1) `bytes.Buffer`
2) `xxhash.Digest`
3) `fnv.Hash32`
None of these 3 can return an error when you look at their code, they are all hard-coded to only return `nil` as error. So I'm not sure if we really have to check for an error here. I wouldn't like to modify `NameWithTags()` so it returns an error, because then every location that uses it would also have to check for that error again etc...

Reply:
writeSortedTagString() can't make assumptions about whether or not the passed in io.Writer actually will return an error or not, so it must check explicitly. (the current implementation does this right), and also return an error via its signature (regardless or whether it truly will at runtime or not)

I think the right solution here is that callers of writeSortedTagString() can choose to ignore the error if they know it'll never happen, which enables NameWithTags to not have to return an error in its signature. However:
1) we should be explicit / a bit more clear in the code. this seems to compile fine:
```
-       writeSortedTagString(nameWithTagsBuffer, m.Name, m.Tags)
+       _ = writeSortedTagString(nameWithTagsBuffer, m.Name, m.Tags)
```
2) we can obviously only do this at callsites where we know the io.Writer won't actually return an error

Reply:
as mauro points out, we only pass those 3 types of values into writeSortedTagsString and none of them returns errors.
this enables to change the signature of PartitionID to not return an error, at least if the caller promises to always provide a valid PartitionByMethod,.. i guess for now we can leave as is and revise later

Reply:
This makes sense to me